driver-core/src/main/java/com/datastax/driver/core/GettableByIndexData.java

```diff
@@ -413,4 +413,35 @@
      * @throws IndexOutOfBoundsException if {@code i} is not a valid index for this object.
      */
     public Object getObject(int i);
+
+    /**
+     * Returns the the {@code i}th value converted to the given Java type.
+     * A suitable {@link TypeCodec} instance for {@code targetClass} must
+     * have been previously registered with the {@link CodecRegistry} currently in use.
+     *
+     * @param i the index to retrieve.
+     * @param targetClass The Java type the value should be converted to.
+     * @return the value of the {@code i}th value converted to the given Java type.
+     * If the CQL value is {@code NULL}, then {@code null} is returned.
+     * @throws com.datastax.driver.core.exceptions.CodecNotFoundException
+     * if no {@link TypeCodec} instance for {@code targetClass} could be found
+     * by the {@link CodecRegistry} currently in use.
+     */
+    <T> T getObject(int i, Class<T> targetClass);
```

Comment:
Since in the Class and TypeToken instance we are providing token information, what do you think about naming these methods (and the symettric setObject methods) as just 'get' and 'set'?